src/main_app/public/api_routes.py

```
@@ -29,16 +29,20 @@ def __init__(self, bp: Blueprint) -> None:
+            ("/templates", "GET", self.templates_list),
+            ("/templates/<string:filter>", "GET", self.templates_list),
+            ("/templates-mismatched-years", "GET", self.templates_mismatched_years_list),
+            ("/templates-need-update", "GET", self.templates_need_update_list),
+            ("/owidcharts/", "GET", self.owid_charts_list),
+            ("/owidcharts/<string:template_filter>", "GET", self.owid_charts_list),
+            ("/languages/<path:file_name>", "GET", self.file_languages),
+        ]
+        for rule, method, target in routes:
+            self.bp.route(rule, methods=[method])(target)
+
+    def templates_list(self, filter: str = ""):
```

Ruff (0.16.1)

[error] Function argument `filter` is shadowing a Python builtin

(A002)